### ISSUE TITLE
The program editor cursor is off by a few lines (Chrome on MacOS)

### ISSUE THREAD

**Issue by pradeeban** [ControlCore-Project/concore-editor#45]
Just open any long program and go to the end.
Click there, and you will find the cursor is off by a couple of lines. 
This is better visible when you click over the lines and highlight them.
This is confusing to the users. Is it because of the syntax highlighting module? If we can get rid of this bug by removing the syntax highlighting, that can be a better workaround.
 
![Screenshot](https://github.com/ControlCore-Project/concore-editor/tree/main/figures/program-editor.png)

**Comment by Aviral09** [comment]
This is a coding logic bug which can be resolved with some debugging. No need to remove syntax highlighting for this will send a PR for the same shortly.

**Comment by pradeeban** [comment]
great. Looking forward to the PR. Thanks.

**Comment by Aviral09** [comment]
I could not replicate this issue on the local machine. Please check if PR #47 fixes this as well.

**Comment by pradeeban** [comment]
This issue is still present. To reproduce, just open the concore editor https://controlcore-project.github.io/concore-editor/ 

Then upload the concore folder from the concore repository and open the "concoredocker.py" (or any sufficiently long file actually), in Chrome.
Then select all the text of the file, or just go to the last line. You start to see how the lines are off and you won't be able to figure out where you are actually typing.

I have included a screenshot in the original report of this bug. You can open it to see how it looks like.

**Comment by pradeeban** [comment]
Even after the last PR merge, the issue still exists.

**Comment by pradeeban** [comment]
We have observed this issue on Chrome on Mac OS.
The issue does not exist in Firefox on Mac OS.

**Comment by pradeeban** [comment]
Chrome on Windows does not have the cursor misaligned the way it is on macOS. So, this can be tricky to reproduce and fix.

**Comment by Aviral09** [comment]
Could you please share the chrome version info and check if any fonts "Arial", "Helvetica", or sans-serif are present?
This can be checked by going into browser settings.

Most likely, this is a font/ configuration issue, as I checked with a friend having mac and this issue doesn't appear. 
The Chrome on macOS he tested had - Version 107.0.5304.87 (Official Build) (x86_64)

**Comment by Aviral09** [comment]
Upon digging further, I found this [issue](https://github.com/react-simple-code-editor/react-simple-code-editor/issues/55).
It clears that this is a browser issue and happens only on Chrome on macOS.

I will spend some time looking for a workaround to fix this. Otherwise, I would change the library used for syntax highlighting.

**Comment by pradeeban** [comment]
Great found. Thanks.

**Comment by parteekcoder** [comment]
hey @pradeeban , can you please tell me about this bug some details , not able to properly understand it from above conversation , so that I will start work upon it

**Comment by pradeeban** [comment]
Hi @parteekcoder,
Please you open this screenshot ->> https://github.com/ControlCore-Project/concore-editor/blob/main/figures/program-editor.png

You will see the curser lines being off by several lines, making the program editor unusable on Chrome on MacOS. This is bad because Chrome is our officially supported browser, and this bug makes the program editor unusable altogether on MacOS.

Please let me know if that makes it clear.

@Aviral09 may have more pointers.

**Comment by parteekcoder** [comment]
> Hi @parteekcoder, Please you open this screenshot ->> https://github.com/ControlCore-Project/concore-editor/blob/main/figures/program-editor.png
> 
> You will see the curser lines being off by several lines, making the program editor unusable on Chrome on MacOS. This is bad because Chrome is our officially supported browser, and this bug makes the program editor unusable altogether on MacOS.
> 
> Please let me know if that makes it clear.
> 
> @Aviral09 may have more pointers.

yeah understood , Is it only happens with large files or small too?

@Aviral09 may have more pointers.  "what do you mean by this"

**Comment by pradeeban** [comment]
It happens with any file. But it will be more obvious in large files.

Appears only in MacOS. This was implemented by @Aviral09. So, I mean, he may be able to give you more guidance if needed (when/if his time permits, that is).

**Comment by parteekcoder** [comment]
> It

OK I will look into this issue

**Comment by pradeeban** [comment]
Great. Thanks.

Please make sure you have access to a Mac OS. It appears only in Mac OS, as far as I am aware.

**Comment by parteekcoder** [comment]
hey @pradeeban @Aviral09 I am not able to reproduce this issue, Are able to reproduce this issue

**Comment by pradeeban** [comment]
Hi @parteekcoder, I can reproduce this even in your own github deployment.

You need MacOS and Chrome to reproduce this.

Open the demo/pm.py and highlight all the text. You will see the issue. The cursor is off by a few lines - especially noticeable in large files.

**Comment by parteekcoder** [comment]
Can you please send a small video

**Comment by parteekcoder** [comment]
It will be difficult to solve until I reproduce it

**Comment by pradeeban** [comment]
It could be an issue that is specific to certain versions of MacOS? Both Mark and I were able to reproduce this. If you cannot reproduce it, please do not attempt a fix.

> ![Screenshot](https://github.com/ControlCore-Project/concore-editor/tree/main/figures/program-editor.png)

**Comment by sairajchouhan** [comment]
hey @pradeeban did you find any solution for this issue?
